community/kernel/src/main/java/org/neo4j/kernel/impl/api/KernelStatement.java

```diff
@@ -108,6 +108,12 @@ public ProcedureCallOperations procedureCallOperations()
     public TokenWriteOperations tokenWriteOperations()
     {
         accessCapability.assertCanWrite();
+
+        if ( !transaction.mode().allowsWrites() )
```

Comment:
Similar checks already exist in `#dataWriteOperations()` and `#schemaWriteOperations()`. I think it is worth to create a method like `assertWritesAllowed(String message)` at this point.